### DISCUSSION TITLE
PhpAmqpLib\Exception\AMQPIOException     Can not enable keepalive: function socket_import_stream does not exist

### DISCUSSION THREAD

**Discussion by SirousFakhri** [php-amqplib/php-amqplib#1166] [Help wanted]
I have PHP/Laravel Project and I'm using rabbitMQ.
I have set up my project with Docker and here my configuration:

```
php8.1

  rabbit:
    container_name: rabbit_app
    hostname: rabbit_app
    restart: always
    ports:
      - "5672:5672"
      - "15672:15672"
    image: "rabbitmq:3.9-management"
```

Now, when I run ```docker compose up -d``` everything is okay and container is up.
but when I want use one of my commands for queue, I got this error:

```
 PhpAmqpLib\Exception\AMQPIOException 

  Can not enable keepalive: function socket_import_stream does not exist

  at S:\Projects\api\vendor\php-amqplib\php-amqplib\PhpAmqpLib\Wire\IO\StreamIO.php:393
    389▕             throw new AMQPIOException('Can not enable keepalive: tls connection does not support keepalive (#70939)');
    390▕         }
    391▕
    392▕         if (!function_exists('socket_import_stream')) {
  ➜ 393▕             throw new AMQPIOException('Can not enable keepalive: function socket_import_stream does not exist');
    394▕         }
    395▕
    396▕         if (!defined('SOL_SOCKET') || !defined('SO_KEEPALIVE')) {
    397▕             throw new AMQPIOException('Can not enable keepalive: SOL_SOCKET or SO_KEEPALIVE is not defined');

  1   S:\Projects\api\vendor\php-amqplib\php-amqplib\PhpAmqpLib\Wire\IO\StreamIO.php:150
      PhpAmqpLib\Wire\IO\StreamIO::enable_keepalive()

  2   S:\Projects\api\vendor\php-amqplib\php-amqplib\PhpAmqpLib\Connection\AbstractConnection.php:253
      PhpAmqpLib\Wire\IO\StreamIO::connect()

```

with command ```docker exec -it php sh``` and then using ```php -m``` I can see that ```socket extension``` is installed.
here the result of ```php -m```
```
[PHP Modules]
bcmath
readline
redis
...
soap
sockets
sodium
...
```

So, what's my mistake? where did it gone wrong?

**Comment by ramunasd** [comment]
Function `socket_import_stream` is available since PHP version 5.4.0, which is 12 years old release.

You did not provide any info about php setup. Which image is it? How reproduce this issue?

**Reply by lukebakken** [reply]
@sirosfakhri as @ramunasd said, you should be able to provide a way for us to reproduce this issue. Right now you're asking us to _guess_, when it should be easy to provide a docker compose file to demonstrate it.

Please let us know what operating system you are using. I see `S:\` which makes me think you're using Docker on Windows, and that your code may be on a network drive.

Also, did you try to figure this out yourself?

https://www.google.com/search?q=%22function+socket_import_stream+does+not+exist%22

**Comment by SirousFakhri** [comment]
@ramunasd @lukebakken thanks for yours answers.
I just fix it. in my ```.env``` replaced host with container name and it did just worked like a charm.
